src/code/z_actor.c

```diff
@@ -457,439 +1269,3557 @@ f32 Actor_XZDistanceToPoint(Actor* actor, Vec3f* point) {
+}
 
-// This function is very similar to OoT's func_80034A14
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_actor/func_800BD888.s")
+void func_800B8E58(Player* player, u16 sfxId) {
+    if (player->currentMask == PLAYER_MASK_GIANT) {
+        func_8019F170(&player->actor.projectedPos, sfxId);
+    } else {
+        Audio_PlaySoundGeneral(sfxId, &player->actor.projectedPos, 4, &D_801DB4B0, &D_801DB4B0, &D_801DB4B8);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_actor/func_800BD9A0.s")
+void Actor_PlaySfxByPos2(Actor* actor, u16 sfxId) {
+    func_8019F1C0(&actor->projectedPos, sfxId);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_actor/func_800BD9E0.s")
+void func_800B8EF4(GlobalContext* globalCtx, Actor* actor) {
+    u32 sfxId;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_actor/func_800BDAA0.s")
+    if (actor->bgCheckFlags & 0x20) {
+        if (actor->depthInWater < 20.0f) {
+            sfxId = NA_SE_PL_WALK_WATER0 - SFX_FLAG;
+        } else {
+            sfxId = NA_SE_PL_WALK_WATER1 - SFX_FLAG;
+        }
+    } else {
+        sfxId = func_800C9BDC(&globalCtx->colCtx, actor->floorPoly, actor->floorBgId);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_actor/func_800BDB6C.s")
+    func_8019F1C0(&actor->projectedPos, NA_SE_EV_BOMB_BOUND);
+    func_8019F1C0(&actor->projectedPos, sfxId + SFX_FLAG);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_actor/func_800BDC5C.s")
+void func_800B8F98(Actor* actor, u16 sfxId) {
+    actor->sfxId = sfxId;
+    actor->audioFlags &= ~(0x10 | 0x08 | 0x04 | 0x02 | 0x01);
+    actor->audioFlags |= 0x02;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_actor/func_800BDCF4.s")
+void func_800B8FC0(Actor* actor, u16 sfxId) {
+    actor->sfxId = sfxId;
+    actor->audioFlags &= ~(0x10 | 0x08 | 0x04 | 0x02 | 0x01);
+    actor->audioFlags |= 4;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_actor/Actor_Noop.s")
+void func_800B8FE8(Actor* actor, u16 sfxId) {
+    actor->sfxId = sfxId;
+    actor->audioFlags &= ~(0x10 | 0x08 | 0x04 | 0x02 | 0x01);
+    actor->audioFlags |= 0x08;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_actor/func_800BDFC0.s")
+void func_800B9010(Actor* actor, u16 sfxId) {
+    actor->sfxId = sfxId;
+    actor->audioFlags &= ~(0x10 | 0x08 | 0x04 | 0x02 | 0x01);
+    actor->audioFlags |= 0x01;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_actor/func_800BE03C.s")
+void func_800B9038(Actor* actor, s32 timer) {
+    actor->audioFlags &= ~(0x10 | 0x08 | 0x04 | 0x02 | 0x01);
+    actor->audioFlags |= 0x10;
 
-Actor* func_800BE0B8(GlobalContext* globalCtx, Actor* inActor, s16 arg2, u8 arg3, f32 arg4) {
-    Actor* actor = globalCtx->actorCtx.actorList[arg3].first;
+    if (timer < 40) {
+        actor->sfxId = NA_SE_PL_WALK_DIRT - SFX_FLAG;
+    } else if (timer < 100) {
+        actor->sfxId = NA_SE_PL_WALK_CONCRETE - SFX_FLAG;
+    } else {
+        actor->sfxId = NA_SE_PL_WALK_SAND - SFX_FLAG;
+    }
+}
```

Comment:
```suggestion
    if (timer < 40) {
        actor->sfxId = 3;
    } else if (timer < 100) {
        actor->sfxId = 2;
    } else {
        actor->sfxId = 1;
    }
}
```
So this code is really interesting. It turns out that these specific values are not sfxId's and both here and OoT are wrong. (all other uses of `actor->sfxId` are correct). But in this very specific case, this is data that gets sent into the io ports of the music macro language (`Audio_PlaySfxAtPosWithSoundScriptIO()` is the function that it's used for).

Basically, either 0, 1, or 2 (they're offset by 1 so that it passes the `sfxId != 0` check and the +1 is removed later on), those values get sent to io port 6 which is stored in `->soundScriptIO[6]` which then the music macro language can read. Basically, this is setting some kind of parameter for the timer state. And documentation of sequence 0 is needed to understand what that parameter is. It's hard to explain, but this will need be fixed in OoT as well.

This makes me wonder if a union or something is needed for `sfxId` because in this single exception, what is stored here is not a sfxId. All other uses of this DO store sfxId's afaik. Might need more discussion, but this is fine for now.

I would also add a temporary comment for now stating that these are not `sfxIds`

Reply:
I think an union would be a good idea here.
If you have a name for this weird timer/state/parameter, I'll happily add it